layers/headers.go

```diff
@@ -14,25 +14,26 @@ type XHeadersLayer struct {
 	XHeaders map[string]string
 }
 
-func (h *XHeadersLayer) OnRequest(state *httransform.LayerState) error {
+func (h *XHeadersLayer) OnRequest(ctx *layers.Context) error {
 	for k, v := range h.XHeaders {
-		state.RequestHeaders.SetString(k, v)
+		ctx.RequestHeaders.Append(k, v)
 	}
 
-	profile, ok := state.RequestHeaders.GetString("x-crawlera-profile")
-	if ok && (profile == "desktop" || profile == "mobile") {
+	profile := ctx.RequestHeaders.GetLast("x-crawlera-profile").Value()
+	if (profile == "desktop" || profile == "mobile") {
```

Comment:
Since you do not need `ok` here, a more common way of having such if is switch:

```golang
switch profile {
case "desktop", "mobile":
	for _, v := range headersProfileToRemove {
		ctx.RequestHeaders.Remove(v)
	}
}
```

Reply:
Thanks